modules/gapi/src/backends/fluid/gfluidbackend.cpp

```diff
@@ -1243,40 +1243,22 @@ void cv::gimpl::GFluidExecutable::reshape(ade::Graph &g, const GCompileArgs &arg
 // FIXME: Document what it does
 void cv::gimpl::GFluidExecutable::bindInArg(const cv::gimpl::RcDesc &rc, const GRunArg &arg)
 {
-    switch (rc.shape)
-    {
-    case GShape::GMAT:    m_buffers[m_id_map.at(rc.id)].priv().bindTo(util::get<cv::Mat>(arg), true); break;
-    case GShape::GSCALAR: m_res.slot<cv::Scalar>()[rc.id] = util::get<cv::Scalar>(arg); break;
-    case GShape::GARRAY:  m_res.slot<cv::detail::VectorRef>()[rc.id] = util::get<cv::detail::VectorRef>(arg); break;
-    case GShape::GOPAQUE: m_res.slot<cv::detail::OpaqueRef>()[rc.id] = util::get<cv::detail::OpaqueRef>(arg); break;
+    magazine::bindInArg(m_res, rc, arg);
+    if (rc.shape == GShape::GMAT) {
```

Comment:
@rgarnov could you explain the logic behind this?

Reply:
The logic here repeats the logic from `bindInArg`, the only exception is that in case of `GMAT` `cv::Mat` is not placed into the magazine, it's bound to a `Fluid::Buffer` instead. With my changes `cv::Mat` is placed into the magazine first (inside `bindInArg`) and after that it's bound to a `Fluid::Buffer`

Reply:
Gotcha, thanks